File: SrdDatabase/Models/Sacco/Transactions/Transaction.cs
using System;

namespace SrdDatabase.Models.Sacco.Transactions
{
    public class Transaction
    {
        public int Id { get; }

        public int MemberId { get; }

        public string Member { get; }

        public int AccountNumber { get; }

        public DateTime Date { get; }

        public bool IsShares { get; }

        public bool IsContribution { get; }

        public int Amount { get; }

        public int? ReceiptNumber { get; }

        public string Notes { get; }

        public string CreatedBy { get; }

        public string UpdatedBy { get; }

        public Transaction(
            int id,
            int memberId,
            string member,
            int accountNumber,
            DateTime date,
            bool isShares,
            bool isContribution,
            int amount,
            int? receiptNumber,
            string notes,
            string createdBy,
            string updatedBy)
        {
            Id = id;
            MemberId = memberId;
            Member = member;
            AccountNumber = accountNumber;
            Date = date;
            IsShares = isShares;
            IsContribution = isContribution;
            Amount = amount;
            ReceiptNumber = receiptNumber;
            Notes = notes;
            CreatedBy = createdBy;
            UpdatedBy = updatedBy;
        }

        // for Dapper
        public Transaction()
        {
        }
    }
}

File: SrdDatabase/ClientApp/src/models/sacco/transaction.ts
import { PagedParameters, PagedResults } from "../shared";

export interface Transaction {
    id?: number;
    memberId?: number;
    member?: string;
    accountNumber?: number;
    date?: Date;
    isShares?: boolean;
    isContribution?: boolean;
    amount?: number;
    receiptNumber?: number;
    notes?: string;
    createdBy?: string;
    updatedBy?: string;
}

export interface TransactionToSend extends Omit<Transaction, "date"> {
    date?: string;
}

export interface TransactionDetails {
    transaction: Transaction;
}

export interface TransactionParameters extends PagedParameters {
    memberId?: number;
    startDate?: Date;
    endDate?: Date;
    receiptNumber?: number;
    isShares?: boolean;
    isContribution?: boolean;
    notes?: string;
}

export interface TransactionParametersToSend extends Omit<TransactionParameters, "startDate" | "endDate"> {
    startDate?: string;
    endDate?: string;
};

export interface TransactionResults extends PagedResults {
    transactions: Transaction[];
}